scilpy/segment/streamlines.py

```diff
@@ -2,56 +2,48 @@
 
 import itertools
 
+from dipy.io.stateful_tractogram import StatefulTractogram
 from dipy.tracking.metrics import length
-from dipy.tracking.streamline import (set_number_of_points,
-                                      transform_streamlines)
-from dipy.tracking._utils import _mapping_to_voxel
+from dipy.tracking.streamline import set_number_of_points
 from dipy.tracking.vox2track import _streamlines_in_mask
 from nibabel.affines import apply_affine
 import numpy as np
 
 
-def target_line_based(streamlines, target_mask, affine=None, include=True):
+def target_line_based(sft, target_mask):
+    sft.to_vox()
+    sft.to_corner()
     # Copy-Paste from Dipy to get indices
+    # Should always be in vox/center space
     target_mask = np.array(target_mask, dtype=np.uint8, copy=True)
-    lin_T, offset = _mapping_to_voxel(affine)
 
-    streamline_index = _streamlines_in_mask(
-        streamlines, target_mask, lin_T, offset)
+    streamline_index = _streamlines_in_mask(list(sft.streamlines), target_mask,
+                                            np.eye(3), [-0.5, -0.5, -0.5])
 
     target_indices = []
     target_streamlines = []
-    for idx in np.where(streamline_index == [0, 1][include])[0]:
+    for idx in np.where(streamline_index == [0, 1][True])[0]:
         target_indices.append(idx)
-        target_streamlines.append(streamlines[idx])
+        target_streamlines.append(sft.streamlines[idx])
 
     return target_streamlines, target_indices
 
 
 def filter_grid_roi(sft, mask, filter_type, is_not):
-    streamlines = list(sft.streamlines)
-    transfo, _, _, _ = sft.space_attributes
-
     line_based_indices = []
     if filter_type == 'any':
-        _, line_based_indices = target_line_based(
-            streamlines, mask, transfo)
+        _, line_based_indices = target_line_based(sft, mask)
     else:
+        sft.to_vox()
+        sft.to_corner()
+        streamline_vox = sft.streamlines
         # For endpoint filtering, we need to keep 2 informations
         # Could be faster for either end, but the code look cleaner like this
```

Comment:
informations (with an s) does not exist. Two pieces of information. I know. Damn, english!

the code looks